### PR TITLE
Tests for push mode

### PR DESCRIPTION
**Pull request by adityathebe**
Resolves: https://github.com/flanksource/incident-commander/issues/313

### CODE REVIEW

File: cmd/server.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ var Serve = &cobra.Command{
 		eventHandlerConfig := events.Config{
 			UpstreamConf: upstreamConfig,
 		}
-		go events.ListenForEvents(context.Background(), eventHandlerConfig)
+		go events.ListenForEvents(context.Background(), db.Gorm, eventHandlerConfig)
```
Comment (yashmehrotra) [inline cmd/server.go:123]: We're passing db object down to 5 functions, do you think we should create an event handler struct ?

```
eventHandler = EventHandler{db: db.Gorm, ctx: context, config: config}

go eventHandler.ListenForEvents()
````

File: upstream/upstream_test.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+	ginkgo.It("should have transferred all the config relationships", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.ComponentRelationship{})
+	})
+
+	ginkgo.It("should have transferred all the config component relationships", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.ConfigComponentRelationship{})
+	})
+})
+
+func populateMonitoredTables(gormDB *gorm.DB) error {
```
Comment (yashmehrotra) [inline upstream/upstream_test.go:175]: Is there any drawback to calling the populate all tables function from duty ?

File: upstream/suite_test.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
+package upstream
```
Comment (yashmehrotra) [inline upstream/suite_test.go:1]: Can you move this file to the root ?

I also did something similar with rbac, if we use this in root the db setup can be used by all the tests

File: cmd/server.go (hunk 1 above)
Comment (adityathebe) [reply cmd/server.go:123]: Sure, I'd prefer that actually.

File: upstream/upstream_test.go (hunk 2 above)
Comment (adityathebe) [reply upstream/upstream_test.go:175]: I don't think there's any. I initially started with duty's dummy for a quick test but then stuck with it as that worked fine.

Comment (yashmehrotra) [reply upstream/upstream_test.go:175]: Use this function, takes db instance and loads all the dummy data
```
fixtures/dummy/all.go
9:func PopulateDBWithDummyModels(gormDB *gorm.DB) error {
```

File: cmd/server.go (hunk 1 above)
Comment (yashmehrotra) [reply cmd/server.go:123]: great 👍🏽

File: upstream/suite_test.go (hunk 3 above)
Comment (adityathebe) [reply upstream/suite_test.go:1]: Make sense. Although a slight problem would be how the packages (`upstream` & `rbac`) would access the db objects. This is unlike duty where all the tests are written in the same package.

Comment (yashmehrotra) [reply upstream/suite_test.go:1]: We can experiment with importing

**Pull request merged**